src/main/java/com/capitalone/dashboard/client/RestClient.java

```diff
@@ -33,107 +35,107 @@ public RestClient(Supplier<RestOperations> restOperationsSupplier) {
+            headers = null;
+        } else {
+            headers = createHeaders(userInfo.getFormattedString());
         }
 
-        long start = System.currentTimeMillis();
-
-        ResponseEntity<String> response
-                = restOperations.exchange(url, HttpMethod.POST, new HttpEntity<Object>(body, createHeaders(userInfo.getFormattedString())), String.class);
-
-        long end = System.currentTimeMillis();
+        return this.makeRestCallPost(url, headers, body);
+    }
 
-        LOG.info("Time taken to make a post call to "+url+" = "+(end-start));
+    private ResponseEntity<String> makeRestCallGet(String url, HttpEntity entity) throws RestClientException {
 
+        long start = System.currentTimeMillis();
+        ResponseEntity<String> response;
+        HttpStatus status = null;
+        try {
+            response = restOperations.exchange(url, HttpMethod.GET, entity, String.class);
+            status = response.getStatusCode();
+        } catch (HttpStatusCodeException e) {
+            status = e.getStatusCode();
+            throw e;
+        } finally {
+            long end = System.currentTimeMillis();
+            LOG.info("makeRestCall op=GET url=" + url + " status=" + status + " duration=" + (end - start));
+        }
         return response;
     }
 
     public ResponseEntity<String> makeRestCallGet(String url) throws RestClientException {
         if (restOperations == null) { return null; }
 
-        long start = System.currentTimeMillis();
-
-        ResponseEntity<String> response = restOperations.exchange(url, HttpMethod.GET, null, String.class);
-
-        long end = System.currentTimeMillis();
-
-        LOG.info("Time taken to make a get call to "+url+" = "+(end-start));
-
-        return response;
+        HttpEntity entity = null;
+        return this.makeRestCallGet(url, entity);
     }
 
     public ResponseEntity<String> makeRestCallGet(String url, String headerKey, String token) throws RestClientException {
         if (restOperations == null) { return null; }
 
+        HttpEntity entity;
         if (StringUtils.isEmpty(headerKey) || StringUtils.isEmpty(token)) {
-            return makeRestCallGet(url);
+            entity = null;
+        } else {
+            entity = new HttpEntity<>(createHeaders(headerKey, token));
         }
+        return this.makeRestCallGet(url, entity);
 
-        long start = System.currentTimeMillis();
-
-        ResponseEntity<String> response = restOperations.exchange(url, HttpMethod.GET, new HttpEntity<>(createHeaders(headerKey, token)), String.class);
-
-        long end = System.currentTimeMillis();
-
-        LOG.info("Time taken to make a get call to "+url+" = "+(end-start));
-
-        return response;
     }
 
     public ResponseEntity<String> makeRestCallGet(String url, RestUserInfo userInfo) throws RestClientException {
         if (restOperations == null) { return null; }
 
+        HttpEntity entity;
```

Comment:
Can you refactor the code as mentioned in`makeRestCallGet(String url, String headerKey, String token)`

Reply:
Done